### ISSUE TITLE
Add the possibility to have one configuration per model

### ISSUE THREAD

**Issue by gmusa** [mariuswilms/media#58]
i want to create individual settings for each class model GeneratorBehavior.         

  // try to find model configuration
 $filter=Configure::read('Media.filter.' . $Model->alias . '.' . Mime_Type::guessName($file));
 if (empty($filter)) {
     $filter=Configure::read('Media.filter.' . Mime_Type::guessName($file));
 }

**Comment by ReplicaObscura** [comment]
I'm trying to tackle this very problem myself (although I'm using my CakePHP 2.0 port of this plugin).

While model-specific configurations would be a good option, I'm wondering if there isn't a workaround in the meantime. What about just reading the Media.filter config value, modifying it, and writing it back again in your beforeSave() method?

You could always save the original value and write that back again after the operations are complete.

I'm not sure if that's too late in the game for the plugin to pick up on the change or not, but if not it would allow model-specific settings without too much fuss.

**Comment by gmusa** [comment]
This is a great solution, but I like having all settings in the same place. Anyway, it's a very good advice, to avoid modifying the plugin's code.

**Comment by ReplicaObscura** [comment]
I made some very simple changes to GeneratorBehavior.php and Config/core.php in my CakePHP 2.0 version, and it now should support model-specific filters (or, technically, any named filters).

https://github.com/bmcclure/CakePHP-Media-Plugin/commit/ed2b5962908052c3be732103f9f90365e948ec6b
https://github.com/bmcclure/CakePHP-Media-Plugin/commit/21364bff3dbac041d6b2d81128ee0c6db6ddfe59

It's untested as of yet, but similar changes should work for the Cake 1.x version of Media.

Basically, you can pass a 'filter' => 'name' option to the GeneratorBehavior and it will use that key, or it will fall back to the model alias as a key, and as a last resort it will use the 'default' configuration key.

Now the configuration would look like:

```
Configure::write('Media.filter', array(
    'default' => array(
        ... (filter config)
    ),
    'BlogPost' => array(
        ... (filter config)
    ),
    'some_name' => array(
        ... (filter config)
    ),
));
```

I'm also planning on optionally merging the selected filter array with the 'default' filter array so that you can set defaults and then override only specific items on a per-model basis.

Anyway, that's my take on the issue, and the solution I've started to implement in my CakePHP 2.0 fork.

**Comment by ReplicaObscura** [comment]
I'm considering getting even more elegant with the solution, such that a filter configuration could be provided directly to the bahavior which would avoid calling Configure::read at all, or a name could be specified which would use that key from the Media.filter array, or it would fallback to the model alias or the default config after that.

That would allow for greater flexibility, since some people might want to leave the generator behavior configuration to the model itself.

This would also make it even easier to, say, allow the user to select what size images they want, and modify the filter in the beforeSave() method to accomodate the submitted size.

I'll probably continue to enhance these features in my 2.0 fork as well--if there is an interest, I could also make the changes in the 1.x version so they can be merged into the core.

**Comment by ReplicaObscura** [comment]
Just an update for anyone interested, I have the solution I just described fully working in my latest commit to my 2.0 branch, and it's pretty awesome so far!

https://github.com/bmcclure/CakePHP-Media-Plugin/commit/f438a8c1fc5acb4b5dad5b8886666a84b7837eec

**Comment by mariuswilms** [comment]
Implemented in 8fd0829. Thanks a lot for all the suggestions!
